CHANGELOG.md

```diff
@@ -9,34 +9,89 @@ The version headers in this history reflect the versions of Apollo Server itself
+
+- Top-level exports have changed. For example:
+
+  - We no longer re-export the entirety of `graphql-tools` (including `makeExecutableSchema`) from all Apollo Server packages. To continue using them, install [`graphql-tools`](https://www.graphql-tools.com/) or one of its sub-packages yourself.
+  - The `Upload` scalar is no longer exported as part of  dropping built-in support for file uploads.
 - Support for serving an HTML UI has been generalized. While servers in dev mode still default to serving GraphQL Playground (note: this may change before v3.0.0), plugins can define a new `renderLandingPage` hook which returns an HTML page that is served to browsers. The `playground` option to `new ApolloServer` has been removed; customizing Playground or making it run in production mode can be done by installing the new `ApolloServerPluginLandingPageGraphQLPlayground` plugin yourself. To disable Playground, either install the new `ApolloServerPluginLandingPageDisabled` plugin or install any other plugin that implements `renderLandingPage`. Packages no longer export `defaultPlaygroundOptions`, `PlaygroundConfig`, and `PlaygroundRenderPageOptions`. By default, no GraphQL Playground settings are overridden, including the endpoint, which now just defaults to `window.location.href` (with most query parameters removed); this means you typically don't have to manually configure the endpoint.
-- Stopped publishing the deprecated `apollo-server-testing` package. This package is just a wrapper around `server.executeOperation` which you can use directly.
+- Stopped publishing the deprecated `apollo-server-testing` package. This package is just a wrapper around `server.executeOperation`, which you can use directly.
+- `apollo-server-caching`: The test suite helper works differently, and the `TestableKeyValueCache` interface is removed.
+
+#### Changes to Node.js framework integrations
+
+- When using a non-serverless framework integration (Express, Fastify, Hapi, Koa, Micro, or Cloudflare), you now *must* call `await server.start()` before attaching the server to your framework. 
+  * This method was introduced in v2.22 but was optional prior to Apollo Server 3.
+  * This does not apply to the core `apollo-server` library or to _serverless_ framework integrations.
```

Comment:
I don't think "core" is right here: this isn't about `apollo-server-core`, it's about the "batteries-included" `apollo-server` which is a wrapper around `apollo-server-express`.